From 23a298b95ffcdb9fa99d056ba37458d068c02181 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Thu, 10 Oct 2024 18:33:10 +0200
Subject: [PATCH 696/954] fix(stm32mp2): reduce BL2 size on trusted boot
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Needed to gain space for BL31. We can gain this space for BL2 in this
config (reduced by 4kB)

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git e8f5714fb855fa0ff9442461bd3b02c4e412a723 fix(stm32mp2): reduce BL2 size on trusted boot]
Change-Id: I6aed7ce1b383557299606f3df23ce971912562cf
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/410573
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 288362a694..e26ccbd281 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -221,7 +221,7 @@ enum ddr_type {
 #define STM32MP_BL2_SIZE		U(0x0002C000)	/* 176 KB for BL2 */
 #else /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
 #define STM32MP_BL2_RO_SIZE		U(0x00021000)	/* 132 KB */
-#define STM32MP_BL2_SIZE		U(0x0002D000)	/* 180 KB for BL2 */
+#define STM32MP_BL2_SIZE		U(0x0002C000)	/* 176 KB for BL2 */
 #endif /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
 #else
 #if STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER
-- 
2.53.0

